src/cmds/qsub_sup.c

```diff
@@ -1410,32 +1411,50 @@ get_comm_filename(char *fname)
 {
 	char *env_svr = getenv(PBS_CONF_SERVER_NAME);
 	char *env_port = getenv(PBS_CONF_BATCH_SERVICE_PORT);
+	char *psi_var = pbs_get_conf_var(PBS_CONF_SERVER_INSTANCES);
 	int count = 0;
+	char buf[LARGE_BUF_LEN];
+	int len;
+	char hash[SHA_DIGEST_LENGTH];
+	int i;
 
-
-	count = snprintf(fname, MAXPIPENAME, "%s/pbs_%.16s_%lu_%.8s_%.32s_%.16s_%.5s",
+	count = snprintf(fname, MAXPIPENAME, "%s/pbs_%.16s_%lu_%.8s_%.32s_%.16s_%.5s_%s",
 		tmpdir,
 		((server_out == NULL || server_out[0] == 0) ?
 		"default" : server_out),
 		(unsigned long int)getuid(),
 		cred_name,
 		get_conf_path(),
-		(env_svr == NULL)?"":env_svr,
-		(env_port == NULL)?"":env_port
+		env_svr ? env_svr : "",
+		env_port ? env_port : "",
+		psi_var ? psi_var : ""
 		);
 
 	if (count >= MAXPIPENAME) {
-		snprintf(fname, MAXPIPENAME, "%s/pbs_%.16s_%lu_%.8s_%.32s_%.16s_%.5s",
-		TMP_DIR,
-		((server_out == NULL || server_out[0] == 0) ?
-		"default" : server_out),
-		(unsigned long int)getuid(),
-		cred_name,
-		get_conf_path(),
-		(env_svr == NULL)?"":env_svr,
-		(env_port == NULL)?"":env_port
-		);
+		count = snprintf(fname, MAXPIPENAME, "%s/pbs_", TMP_DIR);
+		len = snprintf(buf, MAXPIPENAME, "%.16s_%lu_%.8s_%.32s_%.16s_%.5s_%s",
+			 ((server_out == NULL || server_out[0] == 0) ? "default" : server_out),
+			 (unsigned long int) getuid(),
+			 cred_name,
+			 get_conf_path(),
+			 (env_svr == NULL) ? "" : env_svr,
+			 (env_port == NULL) ? "" : env_port,
+			 psi_var ? psi_var : "");
```

Comment:
Just a question. As server and port information is already present in psi do we still need to add evn_svr and evn_port to the pipe name ?

Reply:
pipe name should be unique based on the value in the psi.